### ISSUE TITLE
While trying to add my Hub Tuya is fails with the Connection to device succeeded but no datapoints found, please try again. Create a new issue and include debug logs if problem persists""

### ISSUE THREAD

**Issue by michaelnoardo** [rospogrigio/localtuya#1277]
## Environment

- Localtuya version:  
- Home Assistant Core version:  
- [] Does the device work using the Home Assistant Tuya Cloud component ? 
- [] Does the device work using the Tinytuya (https://github.com/jasonacox/tinytuya) command line tool ? 
- [] Was the device working with earlier versions of localtuya ? Which one? 
- [] Are you using the Tuya/SmartLife App in parallel ? 

## Steps to reproduce

1.
2. 
3.

## Provide Home Assistant traceback/logs

```
put your log output between these markers
```

**Comment by Paul1865** [comment]
I tried with the 3.2 version instead of 3.3 or 3.4. Did the job.

**Comment by klona** [comment]
In my case, whatever version gives the same result

**Comment by jackb007** [comment]
Same problem here. The LocalTuya interface only gives four options, namely 3.1, 3.2, 3.3, and 3.4. Every single Tuya Zigbee gateway available on Aliexpress is Zigbee 3.0.

**Comment by TomHejret** [comment]
@jackb007 I don't think this is the same version. Version 3.0 in case of the gateway refers to Zigbee version, not Tuya protocol.

**Comment by warrenarnold** [comment]
Don't know if this helps but I had over 50 localtuya devices running connected to 3 access points and randomly I would loose 10 or 20 at a time. 
Tried rebooting access points, repairing with the smartlife app and updating keys, removing and re-adding to localtuya with no improvement.

I remembered some instructions for a baby monitor device once said its not compatible with 5gHz (same as tuya) and **not compatible with any wifi channel over channel 10**. Upon looking into it every time my access points (which were set on Auto) went to channel 11 or above I would loose all the localtuya devices on them, even though they were still connected and worked on the smartlife app.

I manually forced all access points to channels below 10 and all my devices were back online within 30 seconds.

If you are having this issue, try locking your wifi access points to channels 10 and below.

**Comment by OG-algo** [comment]
same problem with ZigBee Gateway

**Comment by tasteditchy** [comment]
Same issue here, tuya smart hub. only want to use as the door sensors i have show up as smoke detectors in HA
